wasmtime-environ/src/module_environ.rs

```diff
@@ -34,6 +34,9 @@ pub struct ModuleTranslation<'data> {
     /// Module information.
     pub module: Module,
 
+    /// Contains a signature-to-types mapping
+    pub module_translation_state: ModuleTranslationState,
```

Comment:
This is also what I did, so I don't think I have any alternate suggestions :-p

The one thing I did differently was store it as an `Option` and never have the uninitialized result of `ModuleTranslationState::new` stored here, only either `None` or `Some(initialized)` where `initialized` is the result returned by `translate_module`. The only reason `ModuleTranslationState::new` is public is to facilitate ease of testing for SpiderMonkey where they don't intend to support multi-value yet. Ideally the only way to get a `ModuleTranslationState` would be via `translate_module`.